### ISSUE TITLE
firefox-127 and zimbra and the bitwarden plugin shows empty emails

### ISSUE THREAD

**Issue by Stricken1670** [bitwarden/clients#9618]
### Steps To Reproduce

Zimbra is a webclient/server combination, which we use at our company.
The combination mentioned in the subject reproduces the problem. Any other combination and emails in zimbra are visible.

### Expected Result

zimbra emails are properly displayed.

### Actual Result

Emails appear empty

### Screenshots or Videos

![screenshot_2024-06-13_11-19-30_619090894](https://github.com/bitwarden/clients/assets/172588781/be295af8-f09e-42d1-888f-7635e6132a25)

### Additional Context

In the firefox debug console we found these messages (obfuscated)
[niks.txt](https://github.com/user-attachments/files/15817120/niks.txt)

### Operating System

Windows, macOS, Linux

### Operating System Version

Any version

### Web Browser

Firefox

### Browser Version

127

### Build Version

2024.4.2

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Stricken1670** [comment]
Thanks for looking into the issue, and taking it seriously: The Zimbra version is 9.0.0_GA_4615 (build 20240415043222)  and only affects people who use the classic interface. I hope this helps you to reproduce the problem.

**Comment by m-milinovsky** [comment]
Same here on Windows 10 machines with Firefox 127.0 with Bitwarden plugin 2024.4.2 and Zimbra 10 - also when using Clasic interface only.

**Comment by stefan0xC** [comment]
Not sure what's causing this but if I disable the option for asking to save and use passkeys in the extension it seems to work
![ask_save_passkeys](https://github.com/bitwarden/clients/assets/509385/0baf044c-a30d-497c-92c1-daf8423876a4)

**Comment by m-milinovsky** [comment]
> Not sure what's causing this but if I disable the option for asking to save and use passkeys in the extension it seems to work again.
I can confirm, that disabling "Ask to save and use passkeys" restores the correct behaviour of the plugin.

**Comment by robsee** [comment]
I have the same problem, and it is also fixed by disabling passkeys.

**Comment by hboetes** [comment]
Alright, that's a good workaround, and it will help the devs narrow down the problem.

**Comment by haeiven** [comment]
Hello,

I'm having the same issue and I can confirm that disabling "Ask to save and use passkeys" is a good workaround for now.

**Comment by WoutResseler** [comment]
Have the same issue and indeed disabling "Ask to save and use passkeys" works as a temporary fix

**Comment by zenone87** [comment]
This still happens with version 2024.6.2. The above workaround works.

**Comment by CtrlAltDestroyy** [comment]
Same issue here with Zimbra 10.0.8_GA_4611 (build 20240412081044), Firefox 127.0.1 and Bitwarden Version: 2024.6.2
Disable "Ask to save and use passkeys" resolved this issue, thx

**Comment by Athar42** [comment]
I can also confirm, like the others, the same behavior.
```
Zimbra 9.0.0_GA_1337 (build 20220211171033)
```

Doing the "disable" stuff workaround is working.

**Comment by Finii** [comment]
Same here, broken on Firefox 127 and 128-nightly, ok with Firefox 119 and Ungoogled Chromium 126.0.6478.61
```
Bitwarden 2024.6.2
Zimbra 9.0.0_GA_4612 (build 20240412104050)
```

In the popup show mail window the mouse turns into a ... how is that called, circle with strikethrough thing:

![image](https://github.com/bitwarden/clients/assets/16012374/aa9bb632-9b11-4813-b088-0494fdc52f52)

Disabling passkey workaround fixes it.

Even if the popup mail window is forced into an ordinary window (via Firefox `browser.link.open_newwindow.restriction` = 0) there is no passkey dialogue visible (which is I assume the problem).

**Comment by syserr0r** [comment]
I use passkeys on other sites, is there any other solution?

At present I'm using Google Chrome for Zimbra access as Chrome does not yet appear affected.

**Comment by dagobert** [comment]
Hi,
I also use Zimbra and suffer from the same problem. The temporary fix works.

I investigated further and found something that might help the developers:

The described problem only happens on HTML emails. On plaintext emails the content is displayed. 
Looking into the source code of the Zimbra page, I found a possible reson. The difference between these two types of emails are that HTML emails are inside an iframe.

![grafik](https://github.com/bitwarden/clients/assets/140021/2ab95f3b-bb3d-437b-9a58-1c104c7d5df8)

I hope that helps to fix the problem. I can give more information if necessary.

**Comment by dagobert** [comment]
I searched more and found the following when using bitwarden with passkeys option activated.

![grafik](https://github.com/bitwarden/clients/assets/140021/9df95943-54e2-4033-9003-8faf5ac3e7be)

As you can see, the Iframe content is empty and only filled with the src tag for the JS file.

**Comment by elGranVB** [comment]
Same issue here with Zimbra 9.0.0_GA_4571 (build 20230921061510), Firefox 127.0.2 and Bitwarden Version: 2024.6.3
Disable "Ask to save and use passkeys" workaround still working

**Comment by dagobert** [comment]
I tested the problem in MS Edge using the newest bitwarden Addon. In Edge everything works fine, no problem there.

**Comment by syserr0r** [comment]
@dagobert the issue appears specific to Firefox - I have no issues with chrome (and presumably chromium based browsers like Edge)

**Comment by CirrusThink** [comment]
The issue still appears to 

> Same issue here with Zimbra 9.0.0_GA_4571 (build 20230921061510), Firefox 127.0.2 and Bitwarden Version: 2024.6.3 Disable "Ask to save and use passkeys" workaround still working

I am using Bitwarden version 2024.7.3 , where is the option to Disable "Ask to save and use passkeys" ? I cant seem to find in anywhere in the extension

**Comment by CirrusThink** [comment]
> Ask to save and use passkeys

Ok for MacOS the option to Disable passkeys for workaround purposes is located in Settings --> Notifications

**Comment by m-milinovsky** [comment]
> > Ask to save and use passkeys
> 
> Ok for MacOS the option to Disable passkeys for workaround purposes is located in Settings --> Notifications

Ditto for Windows Settings - Notifications.

**Comment by LiloBzH** [comment]
I ! Same probleme of course.
Firefox 129 / Extension 2024.7.1 

mail blank with passkey enabled and of if disabled but it's not a solution :'(

no idea for ff ?

**Comment by LeanderFischer** [comment]
Yep, also happens for me with Firefox `129.0.2` and Bitwarden `2024.7.1` on Ubuntu 20.04.. Would be nice to get a solution _with_ passkeys :upside_down_face:

**Comment by dagobert** [comment]
I guess there is no importance to this bug since the community using zimbra and bitwarden is not big enough. Firefox users is already a small user base and combind with zimbra it is even smaller.
I just fear that ignoring this bug might mean ignoring a sloppy implementation of inserting the bitwarden code for insertig passkeys option.

**Comment by otbutz** [comment]
Unless someone provides the Bitwarden developers with a Zimbra server and account for testing, I wouldn't hold my breath.

**Comment by hboetes** [comment]
> Unless someone provides the Bitwarden developers with a Zimbra server and account for testing, I wouldn't hold my breath.

But they haven't responded to the ticket, at all.

**Comment by carlinux** [comment]
> Unless someone provides the Bitwarden developers with a Zimbra server and account for testing, I wouldn't hold my breath.

I could create an account for the developers in our  Zimbra to check the issue . (which I have as well). 
Please devs write me a DM do it so anytime.

**Comment by Zalk0** [comment]
I happen to see my mails again on Zimbra, I don't know if Bitwarden corrected it or if it's an update of something else that fixed it.
I have Firefox 130.0.1 and Bitwarden 2024.9.0.

**Comment by LeanderFischer** [comment]
> I happen to see my mails again on Zimbra, I don't know if Bitwarden corrected it or if it's an update of something else that fixed it. I have Firefox 130.0.1 and Bitwarden 2024.9.0.

Yep, I can confirm that on my end. Turned passkeys back on in Bitwarden `Version: 2024.9.0`, using Firefox `129.0.2 (64-bit)` and I can see the mails again in Zimbra `9.0.0_GA_4612 (build 20240412104050)`. This is a newer version of Bitwaden than the one I saw the issue last, so maybe it was fixed :shrug:

**Comment by schewara** [comment]
Not so much luck here, with

Firefox: 130.0 (Linux Flatpak)
Bitwarden: 2024.9.1
Zimbra: 9.0.0_GA_4597 (build 20240217013755)

**Comment by ggrzeczkowicz** [comment]
I also still have the problem on my side. As a big passkey user, it's a real handicap.

I have another workaround to suggest: I've found that Zimbra's modern web client doesn't have this problem, so you can try switching clients in the top right-hand corner of zimbra, in the menu with your username.

Firefox : 131.0.3
Bitwarden extension : 2024.10.0
Zimbra : 9.0.0_GA_4653 (build 20240816023044)

**Comment by McNutnut** [comment]
Still continuing to experience this issue across Zimbra web clients.  Workaround still is to disable passkey offering in Bitwarden, which are becoming more and more prevalent/used as time goes on.

Firefox: 133.0 (Win 11 x64)
Extension: 2024.11.2

**Comment by dagobert** [comment]
I can tell, that after I had problems with Zimbra Version 9 I do not have any problems with Zimbra Version 10.

**Comment by m0rp30** [comment]
Same issue and same solution unflag "Ask to save and use passkeys" in Settings->Notifications

- Zimbra: 8.8.15_GA_4652.FOSS
- Firefox: 134.0.2
- Bitwarden plugin: 2024.12.4

**Comment by andres-asm** [comment]
This is still happening, mitigation sucks because it's global so it's not a solution at all.
Someone closed the issue here: https://community.bitwarden.com/t/firefox-extension-rendering-ajax-webmail-client-unuseable/68565/13

Which sucks too

**Comment by dagobert** [comment]
They probably will not care. You need to update Zimbra to version 9 or 10. With version 10 it is definitely working. The older version will be out of maintainance soon anyway.

**Comment by andres-asm** [comment]
And as you may be well aware, many organizations are stuck with the latest OSS release despite being unsupported.

**Comment by derritter88** [comment]
@dagobert I am running Zimbra 10.1.6 (self compiled) and the issue is still there.

**Comment by pagiraud** [comment]
My organization (employer) is running Zimbra 10.1.6 and I am running Firefox 138.0.1 I can ensure you the issue is still there.

**Comment by twouters** [comment]
This is still broken on Zimbra 10.1.8 OSS with Firefox 139 and Bitwarden 2025.5.0

**Comment by moutaouz** [comment]
I'm encountering the same issue, and as for the others it is fixed by disabling "Ask to save and use passkeys".

**Comment by andres-asm** [comment]
This was working again until last week now it stopped working again

**Comment by NewLunarFire** [comment]
The fix for this issue is creating errors in the console in Firefox. The error is "Uncaught ReferenceError: chrome is not defined" and a comment in the code points back to this issue. The error originates in this script: [](https://github.com/bitwarden/clients/blob/a8d6ad4db6c94bf4b8d320871cb09f746962b11c/apps/browser/src/autofill/fido2/content/fido2-page-script-delay-append.mv2.ts), which points back to this issue.

**Comment by netbrothers-tr** [comment]
I'm late to the party and didn't know anything about the Zimbra issue but can confirm the behaviour described by @NewLunarFire. In my Firefox console I see the following error (twice upon loading any page), only (!) when the BitWarden extension is enabled.

```
Uncaught ReferenceError: chrome is not defined
    <anonymous> <anonymous code>:11
    <anonymous> <anonymous code>:29
    <anonymous> <anonymous code>:31
<anonymous code>:11:5
    <anonymous> <anonymous code>:11
    <anonymous> <anonymous code>:29
    <anonymous> <anonymous code>:31
    #injectIntoMainWorld resource://gre/modules/ExtensionContent.sys.mjs:728
    inject resource://gre/modules/ExtensionContent.sys.mjs:659
    injectInto resource://gre/modules/ExtensionContent.sys.mjs:518
    loadContentScript resource://gre/modules/ExtensionProcessScript.sys.mjs:411
```

**Comment by stefan0xC** [comment]
@NewLunarFire @netbrothers-tr That's a different issue with `browser-v2025.11.1` (cf. https://github.com/bitwarden/clients/issues/17598) and I think not related to this one (since this happened with earlier versions too).

**Comment by netbrothers-tr** [comment]
Thanks, @stefan0xC, for pointing to #17598. That's the correct corresponding issue. I didn't find it, because it's already closed. @NewLunarFire, it seems we have to wait until the patched version gets reviewed and rolled out by Mozilla.

**Comment by andrieshiemstra** [comment]
> I'm late to the party and didn't know anything about the Zimbra issue but can confirm the behaviour described by [@NewLunarFire](https://github.com/NewLunarFire). In my Firefox console I see the following error (twice upon loading any page), only (!) when the BitWarden extension is enabled.
> 
> ```
> Uncaught ReferenceError: chrome is not defined
>     <anonymous> <anonymous code>:11
>     <anonymous> <anonymous code>:29
>     <anonymous> <anonymous code>:31
> <anonymous code>:11:5
>     <anonymous> <anonymous code>:11
>     <anonymous> <anonymous code>:29
>     <anonymous> <anonymous code>:31
>     #injectIntoMainWorld resource://gre/modules/ExtensionContent.sys.mjs:728
>     inject resource://gre/modules/ExtensionContent.sys.mjs:659
>     injectInto resource://gre/modules/ExtensionContent.sys.mjs:518
>     loadContentScript resource://gre/modules/ExtensionProcessScript.sys.mjs:411
> ```

Same issue here, in firefox
